src/BlazorAdmin/Services/HttpService.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace BlazorAdmin.Services
+{
+    public class HttpService
+    {
+        private readonly HttpClient _httpClient;
+        private readonly string _apiUrl;
+
+        public HttpService(HttpClient httpClient, string apiUrl)
+        {
+            _httpClient = httpClient;
+            _apiUrl = apiUrl;
+        }
+
+        public async Task<T> HttpGet<T>(string uri)
+            where T : new()
+        {
+            var result = await _httpClient.GetAsync($"{_apiUrl}{uri}");
+            if (!result.IsSuccessStatusCode)
+            {
+                return new T();
+            }
+
+            return await FromHttpResponseMessage<T>(result);
+        }
+
+        public async Task<T> HttpDelete<T>(string uri, int id)
+            where T : new()
+        {
+            var result = await _httpClient.DeleteAsync($"{_apiUrl}{uri}/{id}");
+            if (!result.IsSuccessStatusCode)
+            {
+                return new T();
```

Comment:
If I try to delete a `CatalogItem` and it fails for some reason, why am I returned a `new CatalogItem()` ?

Reply:
Done